src/base/LruMap.h

```diff
@@ -43,7 +56,7 @@ template <class EntryValue, size_t EntryCost = sizeof(EntryValue)> class LruMap
     /// Search for an entry, and return a pointer
     EntryValue *get(const char *key);
     /// Add an entry to the map
-    bool add(const char *key, EntryValue *t);
+    bool add(const char *key, EntryValue *t, int ttl);
```

Comment:
>  Several of the callers of add() do not have access to the original information used to generate the LruMap

True but irrelevant. LruMap code supporting those callers can and should rely on the already configured `retention` value. The callers do not have to and should not know anything about that aspect of `add()` implementation.

> I can accept this latest proposal, but it will need to do that same mapping defaultTtl() does to avoid negative retention values (infinite freshness) being confused with already-stale negative TTL values.

Glad we found a mutually acceptable solution.

I consider this particular problem addressed.